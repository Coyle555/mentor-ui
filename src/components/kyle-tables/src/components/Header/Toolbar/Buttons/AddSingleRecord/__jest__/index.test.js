import React from 'react';
import { AddSingleRecord } from '../index';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from '@testing-library/react';

afterEach(cleanup);

test('Add single record button default render is correct', () => {
	const component = renderer.create(
		<AddSingleRecord />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Add single record button disabled render is correct', () => {
	const component = renderer.create(
		<AddSingleRecord disabled={true} />
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Add single record button returns the correct data on a callback', () => {
	function onClick(insertType) {
		expect(insertType).toBe('single');
	}

	const { container } = render(
		<AddSingleRecord onClick={onClick} />
	);

	fireEvent.click(container.querySelector('button'));
});
