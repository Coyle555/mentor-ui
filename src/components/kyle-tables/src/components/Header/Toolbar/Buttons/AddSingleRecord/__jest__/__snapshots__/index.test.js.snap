// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add single record button default render is correct 1`] = `
<span
  data-for="table-tooltip"
  data-tip="Add Record"
>
  <button
    className="btn-table"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fal fa-file-plus"
    />
  </button>
</span>
`;

exports[`Add single record button disabled render is correct 1`] = `
<span
  data-for="table-tooltip"
  data-tip="Add Record"
>
  <button
    className="btn-table"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <i
      className="fal fa-file-plus"
    />
  </button>
</span>
`;
